omero/developers/Java.rst

```diff
@@ -990,6 +990,63 @@ in OMERO. Similar approach can be applied when uploading an image.
         store.close();
     }
 
+Sudo (working within another user's context)
+--------------------------------------------
+
+The next example shows how one can work within another user's context. For example
+a data analyst doing some analysis on behalf of a user and attaching the results
+to the user's data. The important point is that the user will be the owner of
+these results and can work with them as usual. The user and analyst don't have to be
+member of a read-annotate group. But the 'analyst' user has to be a 'light admin'
+user with 'sudo' permission.
+
+::
+
+    AdminFacility admin = gateway.getFacility(AdminFacility.class);
+
+    // Look up the experimenter to sudo for
+    ExperimenterData sudoUser = admin.lookupExperimenter(ctx, sudoUserName);
+
+    // Create a SecurityContext for this user within the user's default group
+    // and set the 'sudo' flag (i.e. all operations using this context will
+    // be performed as this user)
+    SecurityContext sudoCtx = new SecurityContext(sudoUser.getGroupId());
+    sudoCtx.setExperimenter(sudoUser);
+    sudoCtx.sudo();
```

Comment:
Is the sudoCtx necessary, I was able to achieve this without setting the analyst as sudo, Please take a look at this:

https://github.com/bramalingam/omero-example-scripts/blob/c06a5517a759ac5069da6508adff334ce95507d0/processing_scripts/Workflow7_ImageJ_Jython_Analyst.py#L138

Reply:
You'd only need the `sudo()` when writing data on behalf of the user, not for just reading it. That might not be very clear in the example, will try to rephrase that a bit maybe.

Reply:
I'll leave that part like it is at the moment. A bit further down in the code snippet there's actually also an example of creating a group context without sudo, that matches more your use case.